tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py

```diff
@@ -463,6 +463,30 @@ def test_absolute_indexing(domain_slice, expected_dimensions, expected_shape):
     assert indexed_field.domain.dims == expected_dimensions
 
 
+def test_absolute_indexing_dim_sliced():
+    domain = common.Domain(
+        dims=(IDim, JDim, KDim), ranges=(UnitRange(5, 10), UnitRange(5, 15), UnitRange(10, 25))
+    )
+    field = common._field(np.ones((5, 10, 15)), domain=domain)
+    indexed_field_1 = field[JDim(8) : JDim(10), : IDim(9)]
+    indexed_field_2 = field[(IDim, UnitRange(5, 9)), (JDim, UnitRange(8, 10))]
```

Comment:
```suggestion
    expected = field[(IDim, UnitRange(5, 9)), (JDim, UnitRange(8, 10))]
```
to emphasize which of the 2 we are testing.
A better unit test would be to test the canonicalization, and test specifically the slicing with the canonical indices. Then this test becomes more like an integration_test.